asserts/database.go

```diff
@@ -176,3 +223,101 @@ func (db *Database) Check(assert Assertion) error {
 	}
 	return fmt.Errorf("failed signature verification: %v", lastErr)
 }
+
+// Add persists the assertions after ensuring it is properly signed and consistent with all the stored knowledge.
+// Returns ErrAssertionNotSuperseding if trying to add a previous revision of the assertion.
+func (db *Database) Add(assert Assertion) error {
+	reg, err := checkAssertType(assert.Type())
+	if err != nil {
+		return err
+	}
+	err = db.Check(assert)
+	if err != nil {
+		return err
+	}
+	primaryKey := make([]string, len(reg.primaryKey))
+	for i, k := range reg.primaryKey {
+		keyVal := assert.Header(k)
+		if keyVal == "" {
+			return fmt.Errorf("missing primary key header: %v", k)
+		}
+		// safety against '/' etc
+		primaryKey[i] = url.QueryEscape(keyVal)
+	}
+	indexPath := filepath.Join(string(assert.Type()), filepath.Join(primaryKey...))
```

Comment:
Not sure about the nesting here.. it makes it convenient to search for items via their primary key order, but makes everything else a pain. That's a detail we can adjust soon, though.. please feel free to move this forward without this being addressed, and let's talk about it at the next opportunity.